test/AsyncSpec.hs

```diff
@@ -21,22 +22,24 @@ spec = describe "async" $ do
       let message :: Member Trace r => Int -> String -> Sem r ()
           message n msg = trace $ mconcat
             [ show n, "> ", msg ]
-
+      ~[lock1, lock2] <- embed $
```

Comment:
ooc, why the lazy match here?

Reply:
Irrefutable pattern match; doesn't require `Sem r` to be `MonadFail`.

Reply:
Right -- makes sense! Thanks for the explanation.